### ISSUE TITLE
Docs: Edits for clarity

### ISSUE THREAD

**Issue by NickolausDS** [globus/django-globus-portal-framework#184]
Some additional changes after #176 to improve clarity and readability.

- [ ] Move "tutorial" docs into distinct section away from "reference" docs
- [ ] Move the following sections out of tutorial and into reference:
    * [Authenticated Search](https://django-globus-portal-framework.readthedocs.io/en/latest/tutorial/search/index-creation.html#authenticated-search)
    * [Multiple Indices](https://django-globus-portal-framework.readthedocs.io/en/latest/tutorial/search/templates.html#advanced-multiple-indices)
- [ ] Add reference ingest document
- [ ] Add reference python script for ingesting search documents

**Comment by bcote-anl** [comment]
In addition to the pull request I just made for the documentation, I would like to discuss additional points.

- [ ] A reference or a link could be added for "See the reference ingest document.." in the note of the Ingesting Metadata section. Otherwise, it is unfortunately unclear where to look.
- [ ] I do like the Ingesting Metadata section and the fact that it highlights the ``url`` key in the ingestion document example. While I understand the ``content`` of a search record is up to the user, I think it would still be nice to have a sentence stating that the ``url`` should be provided in all cases, otherwise the portal will not be able to map the metadata to the actual data.
- [ ] Currently, in terms of transferring data, it is not clear from the documentation that the portal will point to folders, instead of pointing to individual files. Although it is not an issue, it might be important to note this upfront before users start to structure their data and metadata. For example, if someone have thousands of files stored in one single folder in a data collection, the portal could still be used to filter and navigate through the metadata, but the web app transfer would always point to the same single folder regardless of which metadata is selected in the portal.

I would be happy to help working on these points if you think they would improve the documentation.
Thanks for all the hard work you have done already!

**Comment by NickolausDS** [comment]
1. Adding a search reference document is a great idea.
1. I think for most portals (and so far all of our APS portals) linking to files on a Globus collection or some other external file is a common shared pattern. But it's conceivable that someone could create a portal with records that don't have files attached to them. For example, an index could have "User" records with metadata about contributors.
1. This would be good to note if it was a point of confusion. I see the webapp more as a tool to help users navigate to points of interest, but probably isn't a great fit if subjects span thousands of datasets in a single directory. I think a more "productionized" example search ingest document from the first point would be a good place to go over some of the reasons we made certain decisions over others so that users can make informed decisions on whether tehy want to track single files or directories for their search records.

**Comment by bcote-anl** [comment]
Thanks for your responses! 
1. Great.
2. I understand your example with user records, which is a good point, so it might be best to keep the documentation as is and not "force" people to include URLs in their metadata. In addition, the section on fields already implied that the metadata needed the URL in case a portal points to datasets.
3. Sounds good. I see your logic :)
